src/codegen/codegen_cpp_visitor.hpp

```diff
@@ -1409,7 +1417,9 @@ void CodegenCppVisitor::print_vector_elements(const std::vector<T>& elements,
  * different in case of table statement.
  */
 template <typename T>
-void CodegenCppVisitor::print_function_declaration(const T& node, const std::string& name) {
+void CodegenCppVisitor::print_function_declaration(const T& node,
+                                                   const std::string& name,
+                                                   bool hidden) {
```

Comment:
This feels like a point where we'd eventually end up with a lot of arguments. One for: `inline`, `constexpr`, `static`, `extern "C"` and probably a couple of others.

In other places we've noticed that function like `apply(values, f, true, false, true)` (or whatever it was called) aren't extremely easy to read. It's similar here `false` or `true` aren't closely linked to the notion of adding `static`.

Neither is a particular issue; but since it would fix both issues in one go, it might be worth changing the argument `hidden` to something line `modifiers` (or whatever the correct term is) and convert it into an enum.

Reply:
Okay, now I added the option of passing a `std::unordered_set` as the third arg, whose elements are `CppObjectSpecifier`s (basically an enum containing a bunch of common C++ object/function specifiers. Name is subject to change). I contemplated using bit flags, but figured the implementation would be more convoluted, and this part isn't exactly performance- or memory-critical, so I chose to go with a simple set container instead.